cmd/dockerd/daemon.go

```diff
@@ -284,28 +284,32 @@ func (cli *daemonCLI) start(ctx context.Context) (err error) {
 
 	c, err := createAndStartCluster(cli, d)
 	if err != nil {
-		log.G(ctx).Fatalf("Error starting cluster component: %v", err)
+		return errors.Wrap(err, "error starting cluster component")
 	}
 
 	// Restart all autostart containers which has a swarm endpoint
 	// and is not yet running now that we have successfully
 	// initialized the cluster.
 	d.RestartSwarmContainers()
 
-	log.G(ctx).Info("Daemon has completed initialization")
-
-	routerCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
 	// Get a the current daemon config, because the daemon sets up config
 	// during initialization. We cannot user the cli.Config for that reason,
 	// as that only holds the config that was set by the user.
 	//
 	// FIXME(thaJeztah): better separate runtime and config data?
 	daemonCfg := d.Config()
-	routerOpts, err := newRouterOptions(routerCtx, &daemonCfg, d, c)
+
+	bb, sm, err := initBuildkit(ctx, &daemonCfg, d)
```

Comment:
This is the only place where `daemonCfg` is used, so perhaps we should move the code (and comment?) above inside `initBuildkit` as it's already having access to the daemon, and uses it for various other purposes.


The code comment was added in https://github.com/moby/moby/commit/00c9785e2ef52ff96613f7ccc63c784ae6a7fc61 to prevent buildkit being initialised with the wrong "host-gateway-ip";

- https://github.com/moby/moby/pull/47187